libpod/container_internal_linux.go

```diff
@@ -2048,18 +2044,26 @@ func (c *Container) generateResolvConf() (string, error) {
 	return destPath, nil
 }
 
-// generateHosts creates a containers hosts file
-func (c *Container) generateHosts(path string) (string, error) {
+// updateHosts updates the container's hosts file
+func (c *Container) updateHosts(path string) error {
 	orig, err := ioutil.ReadFile(path)
 	if err != nil {
-		return "", err
+		if os.IsNotExist(err) {
+			return nil
```

Comment:
Why return?  Shouldn't this just create a new /etc/hosts with the contents of c.getHosts()?

Reply:
Yes we still have to create /etc/hosts

Reply:
Good catch!